new-log-viewer/src/typings/worker.ts

```diff
@@ -9,6 +9,32 @@ import {
 } from "./logs";
 
 
+/**
+ * Various states of a load process. UI elements may be enabled or disabled based on the state.
+ */
+enum LOAD_STATE {
+
+    /**
+     * When there is no file opened in the viewer.
+     */
+    UNOPENED = "unopened",
+
+    /**
+     * When a file load request is pending response.
+     */
+    LOADING_FILE = "loading",
+
+    /**
+     * When the file is loaded from any request.
+     */
+    READY = "ready",
+
+    /**
+     * When the file is being exported.
+     */
+    EXPORTING = "exporting",
```

Comment:
As discussed offline, this log export state can be too specific. It might be a better idea to generalize this.